ouroboros-consensus/src/Ouroboros/Consensus/Mempool/Impl.hs

```diff
@@ -285,28 +275,30 @@ implTryAddTxs mpEnv wti =
        -> [GenTx blk]
        -> m ([MempoolAddTxResult blk], [GenTx blk])
     go acc = \case
-      []         -> pure (reverse acc, [])
+      []            -> pure (reverse acc, [])
       txs@(tx:next) -> do
         is <- atomically $ takeTMVar istate
-
-        let err :: m ([MempoolAddTxResult blk], [GenTx blk])
-            err = do
-              -- forwarding failed because the changelog anchor changed compared
-              -- to the one in the internal state.
+        mTbs <- getLedgerTablesAtFor ldgrInterface (isTip is) [tx]
+        case mTbs of
+          ChainDB.Tables tbs -> case pureTryAddTxs cfg txSize wti tx is tbs of
+            NoSpaceLeft -> do
               atomically $ putTMVar istate is
-              void $ implSyncWithLedger mpEnv
-              go acc txs
-            ok :: LedgerTables (LedgerState blk) ValuesMK -> m ([MempoolAddTxResult blk], [GenTx blk])
-            ok = \values ->
-              case pureTryAddTxs cfg txSize wti tx is values of
-                NoSpaceLeft             -> do
-                  atomically $ putTMVar istate is
-                  pure (reverse acc, txs)
-                TryAddTxs is' result ev -> do
-                  atomically $ putTMVar istate $ fromMaybe is is'
-                  traceWith trcr ev
-                  go (result:acc) next
-        fullForward mpEnv (isDbChangelog is) [tx] err ok
+              pure (reverse acc, txs)
+            TransactionProcessed is' result ev -> do
+              atomically $ putTMVar istate $ fromMaybe is is'
```

Comment:
I wonder if we should assert somewhere that if `is' == Nothing`, that `result == MempoolTxRejected _ _` and analogously for the case where `is' == Just _`

Reply:
This is already asserted in `pureTryAddTx`